### PR TITLE
SSCS 11213/SSCSCI-6 -Refuse postponement on the day for action postponement request

### PR DESCRIPTION
**Pull request by gemmatalbot**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/SSCSCI-6

### Change description ###

Adding a new drop-down event for postponement called 'Refuse postponement on the day'.

If the postponement request is submitted by the FTA then the FTA state and Interloc state are cleared, and the case satus is set to 'Hearing'

### CODE REVIEW

**Review by nitinprabhuhmcts — APPROVED** [review]

**Review by hashimalisolirius — APPROVED** [review]

**Review by hashimalisolirius — APPROVED** [review]

**Review by nitinprabhuhmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java [hunk 1]
```diff
@@ -174,7 +196,19 @@ private void clearTransientFields(SscsCaseData caseData) {
 
         YesNo unprocessedPostponementRequest = caseData.getPostponementRequest().getUnprocessedPostponementRequest();
         caseData.setPostponementRequest(PostponementRequest.builder()
-            .unprocessedPostponementRequest(unprocessedPostponementRequest)
-            .build());
+                .unprocessedPostponementRequest(unprocessedPostponementRequest)
+                .actionPostponementRequestSelected(caseData.getPostponementRequest().getActionPostponementRequestSelected())
+                .build());
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:201]: are there any other fields in the postponement request that we're clearing here? because for unprocessedPostponementRequest and actionPostponementRequestSelected, we're just setting it to what it was before

File: src/main/java/uk/gov/hmcts/reform/sscs/service/PostponementRequestService.java [hunk 2]
```diff
@@ -23,23 +23,24 @@
 @Service
 public class PostponementRequestService {
 
-    public void processPostponementRequest(SscsCaseData sscsCaseData, UploadParty uploadParty) {
+    public void processPostponementRequest(SscsCaseData sscsCaseData, UploadParty uploadParty, Optional<UploadParty> docUploadParty) {
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/PostponementRequestService.java:26]: why not call the first uploadParty originalSender, like you've done in the other method to remove ambiguity

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToStartHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                .postponementRequest(PostponementRequest.builder()
+                        .actionPostponementRequestSelected("grant")
+                        .build())
+                .build();
+    }
+
+    @Test
+    void givenCaseHasActionPostponementRequestSelected_thenClearActionPostponementRequestSelected() {
+        when(callback.getEvent()).thenReturn(EventType.ACTION_POSTPONEMENT_REQUEST);
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
```
Comment (benouaer) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToStartHandlerTest.java:66]: you can just add caseData to caseDetails

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToStartHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.domain.Hearing;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingOptions;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingRoute;
+import uk.gov.hmcts.reform.sscs.ccd.domain.PostponementRequest;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SchedulingAndListingFields;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsHearingRecordingCaseData;
+
+public class ActionPostponementRequestAboutToStartHandlerTest {
```
Comment (benouaer) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToStartHandlerTest.java:27]: you probably need to test the canHandle method as well, both positive and negative scenarios

**Review by benouaer — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java (hunk 1 above)
Comment (gemmatalbot) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:201]: Yeah there are quite a few postponement fields, I just need the ActionPostponementRequestSelected to remain the same so I can use it later. I reset it to blank in the 'about to start' handler. :)
